### ISSUE TITLE
.gz files are not extracted correctly

### ISSUE THREAD

**Issue by randy3k** [Linuxbrew/brew#767]
To help us debug your issue please explain:
- What you were trying to do (and why)

install [rstudio-server][1] formula

- What happened (include command output)

The .gz resource file is not extracted correctly

see https://circleci.com/gh/randy3k/homebrew-r/418

The formula used to work, so the error is caused by a recent commit.

- What you expected to happen

formula installed correctly

- Step-by-step reproduction instructions (by running `brew` commands)

just tap `randy3k/homebrew-r` and run `brew install rstudio-server`

[1]: https://github.com/randy3k/homebrew-r/blob/rstudio-server-1.1.456%23bottle/Formula/rstudio-server.rb

**Comment by jonchang** [comment]
bisect on linuxbrew says:

```
commit 5b3bbb76c9d224c08ae498677868b69aab71ff7c
Author: Markus Reiter <me@reitermark.us>
Date:   Sun Jul 1 23:35:29 2018 +0200

    Separate staging from download.

:040000 040000 6d5d52263444dbca74d030b9714bd87afff1d386 183e6d00db7777ddcadf1ce9017e320bed2d45c2 M	Library
```

**Comment by jonchang** [comment]
It works on macOS so there must have been a fix applied upstream. This will be addressed once 1.7.2 from upstream is merged and tagged.

**Comment by randy3k** [comment]
It has platform dependent resource files. The macos resource is `.zip` though.

**Comment by jonchang** [comment]
I tested a reduced formula using the .gz file.

**Comment by randy3k** [comment]
Great.

**Comment by sjackman** [comment]
We've recently merged Homebrew/brew into Linuxbrew/brew `master`. Can you test whether this works now?

**Comment by iMichka** [comment]
There is still an issue open regarding an equivalent problem for weird `tgz` files upstream: https://github.com/Homebrew/brew/issues/4648
Just that you know that not everything may be working out of the box.

**Comment by sjackman** [comment]
This issue will be fixed by PR https://github.com/Homebrew/brew/pull/4662, once it's merged into Homebrew/brew and Linuxbrew/brew.

**Comment by sjackman** [comment]
Linuxbrew 1.7.3 includes this fix.
